### PR TITLE
Fix utils for better training

### PR DESCRIPTION
**Pull request by hypnopump**
various fixes for utils which allow AF2 and EGNN to work together and get visually-pleasing results (ESM+sequence embeddings).

good_visuals

### CODE REVIEW

**Comment by hypnopump** [comment]
The output of MDS seems to be nan (dont know why) -> this causes the `sidechain_container` function to fail since it depends on the backbone coordinates to build the sidechain ones.

File: alphafold2_pytorch/utils.py [hunk 1]
```diff
@@ -547,39 +552,66 @@ def sidechain_container(backbones, n_aa, cloud_mask=None, place_oxygen=False,
...
-    if n_aa == 4:
-        new_coords[:, :, 4:] = repeat(predicted[:, :, -1], 'b l d -> b l scn d', scn=10)
-    else:
-        new_coords[:, :, 4:] = repeat(new_coords[:, :, 1], 'b l d -> b l scn d', scn=10)
-    if cloud_mask is not None:
-        new_coords[torch.logical_not(cloud_mask)] = 0.
-    # hard-calculate oxygen position of carbonyl group with parallel version of NERF
-    if place_oxygen: 
-        # build (=O) position of revery aa in each chain
+    with torch.autograd.set_detect_anomaly(True):
```
Comment (lucidrains) [inline alphafold2_pytorch/utils.py:557]: oh oops, was this supposed to make its way into the code?

**Review by lucidrains — COMMENTED** [review]
looks good! one small ask; could you possibly turn on rebasing? https://git-scm.com/book/en/v2/Git-Branching-Rebasing 

you'll just have to run `git config --global pull.rebase true` once!

**Comment by hypnopump** [comment]
Hmm yeah actually not, the issue is fixed, but i plan to make a rearrangement in this part during the next week, so if it's not urgent, i think we can close the PR for now, and i'll redo it once i've got everything assembled.

**Pull request closed without merging**